OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -232,6 +232,12 @@ bool IsBlockedBy(Actor self, Actor otherActor, Actor ignoreActor, CellConditions
 			if (otherActor == ignoreActor)
 				return false;
 
+			// If self is null, we don't have a real actor - we're just checking what would happen theoretically.
+			// In such a scenario - we'll just assume any other actor in the cell will block us by default.
```

Comment:
See `CanMoveFreelyInto` - if there's a free sub cell you're fine. Only after that do blockers get checked.